src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -561,7 +583,7 @@ private async Task RefreshKeyValueCollections(CancellationToken cancellationToke
                         });
                     }
 
-                    keyValueChanges = await _client.GetKeyValueChangeCollection(
+                    keyValueChanges = await client.GetKeyValueChangeCollection(
                         currentKeyValues,
                         new GetKeyValueChangeCollectionOptions
                         {

```

Comment:
Below is `UpdateCacheExpirationTime(changeWatcher, success);` still being executed in finally. Shouldn't we move it in to only execute in success case?